examples/moviegen/README.md

```diff
@@ -0,0 +1,348 @@
+|:---------:|:-------------:|:-----------:|:-------------------:|
+|   2.3.1   |   24.1.RC2    | 7.3.0.1.231 |    8.0.RC2.beta1    |
+
+1. Install MindSpore according to the [official instructions](https://www.mindspore.cn/install).
+   For Ascend devices, please install
+   [CANN8.0.RC2.beta1](https://www.hiascend.com/developer/download/community/result?module=cann&cann=8.0.RC2.beta1)
+   and [MindSpore 2.3.1](https://www.mindspore.cn/install).
+2. Install requirements
+    ```shell
+    pip install -r requirements.txt
+    ```
+
+## Model Weights
+
+<details>
+<summary><b>TAE</b></summary>
+
+We use SD3.5 VAE to initialize the spatial layers of TAE since both have the same number of latent channels, i.e., 16.
+
+1. Download SD3.5 VAE from [huggingface](https://huggingface.co/stabilityai/stable-diffusion-3.5-large/tree/main/vae)
+
+2. Inflate VAE checkpoint for TAE initialization by
+    ```shell
+    python inflate_vae_to_tae.py --src /path/to/sd3.5_vae/diffusion_pytorch_model.safetensors --target models/tae_vae2d.ckpt
+    ```
+
+</details>
+
+<details>
+<summary><b>Text Encoders</b></summary>
+
+Downloading and conversion of the text encoders' weights to the `.safetensors` format can be done automatically by using
+the following commands:
+
+```shell
+python tools/download_convert_st.py "google/byt5-small"
+python tools/download_convert_st.py "google/ul2"
+```
+
+If you face an SSL certificate verification error, you can add `--disable_ssl_verify` option.
+
+</details>
+
+## Generating Text Embeddings
+
+Due to the large memory footprint of the text encoders, the inference and training pipelines don't support generating
+text embeddings online. Therefore, you need to prepare them in advance by running the following command:
+
+```shell
+python inference_text_enc.py \
+--model_name google/ul2 \
+--prompts_file /path/to/prompts.csv \
+--output_path /path/to/output/directory \
+--model_max_length 512
+```
+
+> [!NOTE]
+> We use the sequence length of 512 tokens for UL2, 256 for MetaCLIP, and 100 for ByT5.
+
+## Inference
+
+For more detailed instructions, please run `python inference.py --help`.
+
+### Text-to-Image
+
+```shell
+python inference.py \
+--config configs/inference/moviegen_t2i_256x256.yaml \
+--model.name llama-5B \
+--model.pretrained_model_path /path/to/llama-5B.ckpt \
+--text_emb.ul2_dir /path/to/ul2_embeddings \
+--text_emb.metaclip_dir /path/to/metaclip_embeddings \
+--text_emb.byt5_dir /path/to/byt5_embeddings \
+--image_size 256 455 \
+--batch_size 2
+```
+
+### Text-to-Video
+
+```shell
+python inference.py \
+--config configs/inference/moviegen_t2i_256x256.yaml \
+--model.name llama-5B \
+--model.pretrained_model_path /path/to/llama-5B.ckpt \
+--text_emb.ul2_dir /path/to/ul2_embeddings \
+--text_emb.metaclip_dir /path/to/metaclip_embeddings \
+--text_emb.byt5_dir /path/to/byt5_embeddings \
+--image_size 256 455 \
+--num_frames 32 \
+--batch_size 2 \
+--save_format mp4
+```
+
+### TAE
+
+#### Encoding Video
```

Comment:
this part is not important for users. better fold it by default or just remove it (it's written in TAE section)